hash_practice/exercises.py

```diff
@@ -2,18 +2,55 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(nk log k)
+        Space Complexity: O(n)
+
     """
-    pass
+    """ not sure why split() doesn't work:
+    # for word in strings:
+    #     split_word ="".join(word.split("").sort()) #aet
+    """
+    new_group_words = []
+    word_dict = {}
+
+    for word in strings:
+        split_word = []
+        for char in word:
+            split_word.append(char)
+            split_word.sort()

```

Comment:
Note that because this `split_word.sort()` is in the `for char in word:` loop, it happens on every iteration, which actually bumps your time complexity even higher than your calculation.